docs/api.md

```diff
@@ -0,0 +1,193 @@
+## Getting Things
+
+Retrieving things from the API happens usually with the following call.  We use `await` to wait for the promised value before proceeding:
+
+```js
+await paper.get(...)
+```
+
+## Canvas for Paper
+
+This gives you a canvas to draw things over a specific paper:
+
+```js
+const canvas = await paper.get('canvas');                   // get my paper's canvas
+const canvas = await paper.get('canvas', { width: 500 });   // get canvas, with custom size
+const canvas = await paper.get('canvas', { number: 1234 }); // get another paper's canvas
+```
+
+## Canvas for Projector (a.k.a. Supporter)
+
+This gives you a canvas over the entire projection.  (Each paper gets its own, so no interference)
+
+```js
+const supporterCanvas = await paper.get('supporterCanvas');
+```
+
+## Get Paper Data
+
+All papers are assigned a unique number.  To get your paper's number:
+
+```js
+const number = await paper.get('number');
+```
+
+This number can be used to lookup information about the paper:
+
+```js
+const papers = await paper.get('papers');
+
+papers[number].points   // corners of the paper {center,topLeft,topRight,bottomRight,bottomLeft}
+papers[number].markers  // any dots detected inside the paper (array of {position,color})
+papers[number].data     // arbitrary data set by paper.set('data')
+```
+
+## Setting Paper Data
+
+You can set attributes on your paper that other papers can use:
+
+```js
+await paper.set('data', { someKey: 'someValue' });
+```
+
+## Cover Paper in an Iframe
+
+You can set your paper to display an iframe:
+
+```js
+await paper.set('iframe', { src: 'http://www.example.com' });
+```
+
+## Paper Whisker to detect nearby papers
+
+A paper can be given a "whisker" that lets you know when it touches another paper:
+
+```js
+paper.whenPointsAt({
+  callback: ({paperNumber, paperObj}) => {
+    /* do something */
+  },
+
+  // optional
+  direction,      // "up" (default), "down", "left", "right"
+  whiskerLength,  // as fraction of the side (default 0.7)
+  requiredData,   // array of data fields that must be present in the other paper
+  paperNumber,    // paper number to do this for (default is own paper number)
+});
+```
+
+## Camera Access
+
+You can draw a region of the camera picture to another destination region (using arbitrary quadrilaterals).
+
+For example, to draw another paper's picture to your paper:
+
+```js
+const papers = paper.get('papers');
+const camera = paper.get('camera');
+paper.drawFromCamera(ctx, camera, papers[someNum].points, papers[myNum].points);
+```
+
+## Marker Points
+
+Any dot that is not detected as part of a paper's corner is exposed as a _marker_.
+
+All detected markers are exposed here:
+
+```js
+const markers = paper.get('markers');
+```
+
+Each marker has the following properties:
+
+```js
+{
+  position: {x, y},  // global position
+  color: [r, g, b]
+  colorName // "red", "green", "blue" or "black"
+
+  // these properties will be null if the marker is not on a paper
+  paperId,
```

Comment:
Only noticing this now, but should we call this `paperNumber` for consistency? (This is outside the scope of this PR though, and no biggie anyway)

Reply:
ah, I pulled this from https://github.com/janpaul123/paperprograms/pull/43#discussion_r182001387, but it looks like it was not implemented.  fixed the name, and tagged as wip.